### PR TITLE
#9965 Feature/locale in sylius install

### LINKED ISSUE #9965: [Installer] Include locale question in sylius:install command
Currently, during the installation process (exactly in the `sylius:install:setup` command), the user is asked only about a base currency of the created channel. Maybe we should also ask for the default locale? Having `en_US` as a default locale leads to issues like https://github.com/Sylius/Sylius/issues/9960.

I believe there can be a problem how we should then set up the base application locale defined in `config/services.yaml`.

### PR DESCRIPTION
**Pull request by oallain**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | fixes #9965 
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/CoreBundle/Installer/Setup/LocaleSetup.php [hunk 1]
```diff
@@ -41,27 +43,53 @@ public function __construct(RepositoryInterface $localeRepository, FactoryInterf
     /**
      * {@inheritdoc}
      */
-    public function setup(InputInterface $input, OutputInterface $output): LocaleInterface
+    public function setup(InputInterface $input, OutputInterface $output, QuestionHelper $questionHelper): LocaleInterface
     {
-        $name = $this->getLanguageName($this->locale);
+        $name = $this->getLanguageCodeFromUser($input, $output, $questionHelper);
```
Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/Installer/Setup/LocaleSetup.php]: Could you rename it to `$code`?

**Review by pamil — APPROVED** [review]

File: src/Sylius/Bundle/CoreBundle/Installer/Setup/LocaleSetup.php (hunk 1 above)
Comment (oallain) [reply src/Sylius/Bundle/CoreBundle/Installer/Setup/LocaleSetup.php]: Yes, of course :smile:

**Review by GSadee — APPROVED** [review]

**Review by lchrusciel — APPROVED** [review]

**Comment by Zales0123** [comment]
I tried to check it out before the merge and I realized some strange bug - I believe it's related to the `LanguageBundle`, not proposed changes. The thing is, a translator does not accept compound locales codes (like `pl_PL`, `de_DE`, `jp_JP`), it works only with few of them (as `en_US`) 😕 

Zrzut ekranu 2019-04-1 o 16 37 42

@oallain can you check it out? We should not allow such behaviour, especially as I think these two-parts codes are *the right ones* 🚀 Thank you in advance!

**Comment by oallain** [comment]
Hello,

Yes, i'll loke that :eyes:

**Comment by oallain** [comment]
Hi @Zales0123,

I missed that, `Intl::getLanguageBundle()->getLanguageName($language, $region)` has 2 arguments, it was the cause of the malfunction.

I have now fixed the problem. :smile: 

Regards

**Review by lchrusciel — APPROVED** [review]

**Review by GSadee — APPROVED** [review]

File: src/Sylius/Bundle/CoreBundle/Installer/Setup/LocaleSetup.php [hunk 2]
```diff
@@ -41,29 +43,62 @@ public function __construct(RepositoryInterface $localeRepository, FactoryInterf
...
+        return trim($questionHelper->ask($input, $output, $question));
+    }
+
     private function getLanguageName(string $code): ?string
     {
-        return Intl::getLanguageBundle()->getLanguageName($code);
+        $language = $code;
+        $region = null;
+
+        if (count(\explode('_', $code)) === 2) {
```
[the remark this answers is not in the document]
Comment (lchrusciel) [reply src/Sylius/Bundle/CoreBundle/Installer/Setup/LocaleSetup.php]: PHPStorm recommend having `\` for performance improvements

[the remark this answers is not in the document]
Comment (Zales0123) [reply src/Sylius/Bundle/CoreBundle/Installer/Setup/LocaleSetup.php]: I suppose it won't matter in such a simple case xD and is inconsistent with other's usage of such a function in Sylius. But ☮️

[the remark this answers is not in the document]
Comment (oallain) [reply src/Sylius/Bundle/CoreBundle/Installer/Setup/LocaleSetup.php]:
```suggestion
        if (count(explode('_', $code, 2)) === 2) {
```

File: src/Sylius/Bundle/CoreBundle/Installer/Setup/LocaleSetup.php [hunk 3]
```diff
@@ -41,29 +43,62 @@ public function __construct(RepositoryInterface $localeRepository, FactoryInterf
...
+            [$language, $region] = \explode('_', $code);
```
Comment (oallain) [inline src/Sylius/Bundle/CoreBundle/Installer/Setup/LocaleSetup.php]:
```suggestion
            [$language, $region] = explode('_', $code, 2);
```

**Comment by oallain** [comment]
@Zales0123, i think it's now better :smile:

**Review by Zales0123 — APPROVED** [review]

**Comment by Zales0123** [comment]
Thanks, @oallain! :tada:

**Comment by elboletaire** [comment]
This does not seem to be editing `config/services.yaml`. After a fresh install of sylius setting a different language other than `en_US` I had to manually change the `services.yaml` file, as it was showing `en_US`.

**Pull request merged**